apis/apiextensions/v2/conditions.go

```diff
@@ -31,16 +31,19 @@ const (
 
 	// A TypeOffered XRD has created the CRD for its composite resource claim
 	// and started a controller to reconcile instances of said claim.
+	// Deprecated: Claims aren't supported in apiextensions.crossplane.io/v2.
```

Comment:
Have to admit that AI added these deprecated warnings on Claim related fields, but left it as it makes sense. 

The deprecations cause a lot of `//nolint:staticcheck // we are still supporting v1 XRD` but I think its worth it.